include/drivers/uart.h

```diff
@@ -486,8 +486,8 @@ static inline int uart_callback_set(const struct device *dev,
  * @param dev     UART device structure.
  * @param buf     Pointer to transmit buffer.
  * @param len     Length of transmit buffer.
- * @param timeout Timeout in milliseconds. Valid only if flow control is
```

Comment:
Open question: should we use `int64_t` to allow for longer timeouts?

Reply:
I think there is no need to. With `int32_t`, the maximum value gives almost 36 minutes, so it seems to be enough for UART communication.